### PR TITLE
feat: pdm init template argument

### PR DESCRIPTION
**Pull request by frostming**
Signed-off-by: Frost Ming <me@frostming.com>

## Pull Request Checklist

- [ ] A news fragment is added in `news/` describing what is new.
- [ ] Test cases added for changed code.

## Describe what you have changed in this PR.

Discussion: #2037

### CODE REVIEW

**Comment by frostming** [comment]
An initial working implementation is ready, you can test it now with `git+https://github.com/pdm-project/pdm@feat/template`
@pawamoy @larrycai

File: src/pdm/cli/templates/default/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[project]
+name = "example-package"
+version = "0.1.0"
+description = "Default template for PDM package"
+authors = []
+dependencies = []
+requires-python = ">=3.7"
```
Comment (pawamoy) [inline src/pdm/cli/templates/default/pyproject.toml:7]: I guess you could use 3.8 since 3.7 is EOL in two days.

Comment (frostming) [reply src/pdm/cli/templates/default/pyproject.toml]: Updated. This value is a placeholder and will be replaced by what the users input.

**Comment by larrycai** [comment]
````
Username for 'https://github.com': larrycai   
Password for 'https://larrycai@github.com': 
````

Why do we need to ask for user&passwd, is it anonymous access?

**Comment by larrycai** [comment]
Some testing

## name 

````
$ pdm init gitlab/copier-fastapi => https://github.com/pdm-project/template-gitlab/copier-fastapi/ not found
$ pdm init ../gitlab/copier-fastapi  => check locally, and I have folder there, it works
````

or `if "/" in name`, we treat it is local package?

## template 

````
$ pdm --version
PDM, version 2.7.5.dev32+g9854a9ea
$ pdm init django 
Creating a pyproject.toml for PDM...
Please enter the Python interpreter to use
..
Please select (0): 
Is the project a library that is installable?
If yes, we will need to ask a few more questions to include the project name and build backend [y/n] (n): y
Project name (tmp): hello
...
"/opt/homebrew/Cellar/python@3.11/3.11.4/Frameworks/Python.framework/Versions/3.11/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1126, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1140, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'django_app'
````

**Comment by pawamoy** [comment]
`namespace/proiect` was left out of the feature, yes.

What is `django` in your context? A folder in the current working directory?

**Comment by larrycai** [comment]
> `namespace/proiect` was left out of the feature, yes.
> 
> What is `django` in your context? A folder in the current working directory? Is this the full traceback?

I saw https://github.com/pdm-project/template-django, then just want to give a try ;-). HappyTesting or ChaosTesting

**Comment by frostming** [comment]
@larrycai Thanks for the testing, it was fixed.

The complete logic finding template is here: https://github.com/pdm-project/pdm/blob/3339e8b5252fba01e71706d6a85eb0bd3fbea397/src/pdm/cli/templates/__init__.py#L80-L89

**Comment by frostming** [comment]
> ````
Username for 'https://github.com': larrycai   
Password for 'https://larrycai@github.com': 
````

Why do we need to ask for user&passwd, is it anonymous access?
 

It is just inheriting your current git authentication. make sure to set credentials correctly so you can successfully run `git clone`.

**Comment by frostming** [comment]
Now `pdm init django` should work like a charm!

**Comment by larrycai** [comment]
thx for quick fix, well done!

**Comment by frostming** [comment]
Now `pdm init` also supports `--copier` and `--cookiecutter` option, to use another project generators. They are now listed as optional dependencies, although I don't see any advantage over calling `copier` and `cookiecutter` CLI directly.

**Pull request merged**